Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1

```diff
@@ -1575,6 +1575,13 @@ function New-M365DSCConnection
                 -Credential $InboundParameters.Credential `
                 -SkipModuleReload $Global:CurrentModeIsExport `
                 -ProfileName $ProfileName
+
+            #2942 - check correct profile applied
+            if (($ProfileName -ne $global:MSCloudLoginConnectionProfile.MicrosoftGraph.ProfileName) -and ($Workload -eq 'MicrosoftGraph') )
+            {
+                select-mgprofile -Name $profilename
```

Comment:
Could you update the cmdlet to match the default PascalCase format: Select-MgProfile

And update the variable name to match the earlier used PascalCasing: $ProfileName

Both are true for all instances you have added.